backend/src/main/java/hu/flowacademy/vajdasagbrand/service/ItemService.java

```diff
@@ -24,10 +24,10 @@ public Item createItem(Item item) throws ValidationException {
     }
 
     public Item deleteById(String id) throws ValidationException {
-        Item deleted = findFirstByIdAndDeletedAtNotNull(id);
-        deleted.setDeletedAt(LocalDateTime.now().withNano(0));
-        itemRepository.save(deleted);
-        return deleted;
+        Item toBeDeleted = itemRepository.findFirstByIdAndDeletedAtNull(id).orElseThrow(
+                () -> new ValidationException("No item found with given id"));
+        toBeDeleted.setDeletedAt(LocalDateTime.now().withNano(0));
+        return itemRepository.save(toBeDeleted);
```

Comment:
Nem kell kiszervezned külön változóba vagyis a toBeDeleted-be, mehet a függvény hívás a save-be egyből.

Reply:
javítottam